internalshared/configutil/config.go

```diff
@@ -75,10 +76,12 @@ func ParseConfig(d string) (*SharedConfig, error) {
 	}
 
 	if result.DisableMlockRaw != nil {
-		if result.DisableMlock, err = parseutil.ParseBool(result.DisableMlockRaw); err != nil {
-			return nil, err
+		isDisabled, err := parseutil.ParseBool(result.DisableMlockRaw)
+		if err != nil {
+			return nil, fmt.Errorf("error parsing 'disable_mlock': %w", err)
+		} else if !isDisabled {
+			return nil, fmt.Errorf("OpenBao does not support mlock; please disable and use CGroups instead")
```

Comment:
Probably here as well I'd recommend the option.

Do we want to just always err in this block? I don't _really_ see the value in even attempting to parse the boolean? My 2c.?

Reply:
IMO this really depends on how much OpenBao wants to be a (near) drop-in replacement for Vault 1.14. My thinking was that if `disable_mlock` was absent in a config, that meant the user didn't really consider the issue, so we're fine to do something different. And if mlock was explicitly disabled, we're also fine. But if a user explicitly added a configuration option to say that mlock was enabled, which was the default in Vault anyway, then OpenBao should error out and bring this change in behavior to this user's attention.

Reply:
Ah I see, this is an err. I'd probably warn always (if not erring), and if this is enabled, then error.

However, note that [by the RFC](https://openbao.org/docs/policies/migration/#proposal), we're only aiming for API compatibility.

Reply:
What would be the idiomatic way to warn in `internalshared/configutil`? I don't see any other warnings currently in that file, or in the files that call `configutil.ParseConfig`.

Reply:
Well, I guess the logger isn't setup at this point, so the short answer is there is none :D 

I think that means it our options are to treat it like an error or ignore it... Thoughts? I guess erring only if set to true is probably the kinder thing to do, so let's leave this as is.